osu.Game/Screens/Ranking/Expanded/StarRatingDisplay.cs

```diff
@@ -22,22 +24,37 @@ namespace osu.Game.Screens.Ranking.Expanded
     /// </summary>
     public class StarRatingDisplay : CompositeDrawable
     {
-        private readonly StarDifficulty difficulty;
+        [Resolved]
+        private OsuColour colours { get; set; }
+
+        private CircularContainer colorContainer;
+        private OsuTextFlowContainer textContainer;
+        private CancellationTokenSource cancellationTokenSource;
+        private IBindable<StarDifficulty?> bindableStarDifficulty;
+
+        private readonly StarDifficulty starDifficulty;
+        private readonly BeatmapInfo beatmapInfo;
 
         /// <summary>
         /// Creates a new <see cref="StarRatingDisplay"/> using an already computed <see cref="StarDifficulty"/>.
         /// </summary>
         /// <param name="starDifficulty">The already computed <see cref="StarDifficulty"/> to display the star difficulty of.</param>
         public StarRatingDisplay(StarDifficulty starDifficulty)
         {
-            difficulty = starDifficulty;
+            this.starDifficulty = starDifficulty;
         }
 
-        [BackgroundDependencyLoader]
-        private void load(OsuColour colours, BeatmapDifficultyCache difficultyCache)
+        /// <summary>
+        /// Creates a new <see cref="StarRatingDisplay"/> using a <see cref="BeatmapInfo"/> to use a bindable for the difficulty.
+        /// </summary>
+        /// <param name="beatmapInfo">The <see cref="BeatmapInfo"/> to use to create a bindable for <see cref="StarDifficulty"/></param>
+        public StarRatingDisplay(BeatmapInfo beatmapInfo)
         {
-            AutoSizeAxes = Axes.Both;
+            this.beatmapInfo = beatmapInfo;
+        }
```

Comment:
I agree with you in the regard that the StarRatingDisplay shouldn't actually have a dependency on the beatmapInfo but how would you manage updating the star rating? You could Expire the whole container and add a new one with the updated rating but especially in the BeatmapInfoWedge case you would have the problem of sorting the children so the rating is above the StatusPill

Reply:
Not sure what the problem is there in particular. Isn't everything recreated on every beatmap change as-is anyway?

The way I see it, the front layer (the not-background one) should either get fully recreated (in which case this ceases to be a problem), or fully update dynamically (so it just gets created once in the start and then only data in it changes), which also makes "children sorting" concerns null and void.

Reply:
The star rating gets also updated on mod change therefor the wedge needs to be updated without changing the beatmap. I tried the fully update dynamically way with the beatmapInfo in the constructor. It could also also have a property with the star rating which can get changed outside and updates itself accordingly then

Reply:
Ok, fair point. Would rather have the star rating as a plain public property or bindable on `StarRatingDisplay`, then.